lib/Doctrine/Migrations/DependencyFactory.php

```diff
@@ -281,12 +282,19 @@ public function getMigrationRepository() : MigrationRepository
                 $this->getConfiguration()->getMigrationClasses(),
                 $this->getConfiguration()->getMigrationDirectories(),
                 $this->getMigrationsFinder(),
-                new MigrationFactory($this->getConnection(), $this->getLogger()),
+                $this->getMigrationFactory(),
                 $this->getVersionComparator()
             );
         });
     }
 
+    public function getMigrationFactory() : MigrationFactory
```

Comment:
Is it necessary that this method is public?

Reply:
good point, fixed

Reply:
sorry, i have re exposed it again. users need to fetch the old migration factory and being able to decorate it with their own implementation